### ISSUE TITLE
installer overwrites /etc/exports

### ISSUE THREAD

**Issue by pcnate** [FOGProject/fogproject#187]
Fog 1.4.4 overwrites /etc/exports files without creating a backup.

The expected behavior is one of the following:
 1) warn of overwrite
 2) append to end of file
 3) provide a difference of the file
 4) backup the file

**Comment by Sebastian-Roth** [comment]
Thanks for reporting. There is an installer option for not recreating the exports file: `--no-exportbuild` or `-E`. Or you can edit `/opt/fog/.fogsettings` and set `blexports='0'` to leave this file untouched.

I know most packaging systems do a great job in giving the choices you mentioned. But the installer script was written this way years ago and it would mean re-writing a good part of it to change this as we have this same not just for the exports file but also for dhcp config and others.

Do you know of a great mechanism like debian's `dpkg-divert` that is available on all FOG supported linux systems (debian, ubuntu, redhat, centos, fedora, arch, ...)? If something like `dpkg-divert` exists, then I'd think about changing the installer as this is a lot more powerful and we would be able to revert the configs on uninstall.

**Comment by pcnate** [comment]
Could you simply have a prompt to overwrite or append?

**Comment by mastacontrola** [comment]
Why not just use the `--no-exportbuild` or `-E` flag. It only needs to be done once, and after that your blexports line that was defined by @Sebastian-Roth will be set to 0 for you. Then you don't have to worry about it being overwritten. This was the point of that argument.

**Comment by pcnate** [comment]
I feel that this flag should instead be --no-prompt-export

**Comment by Sebastian-Roth** [comment]
You mean we should rename the installer command line option? Why?

**Comment by pcnate** [comment]
As a safety check. I had to recreate an export file because I installed it on a machine that had exports. Completely forgot about exports when I installed it. When I first installed fog, I had no idea it even accepted arguments. I didn't learn that until recently.

I had done a simple check for my modifications to a file in the past so I could append to a file only once.
https://github.com/pcnate/npm-configure/blob/55764b38487ee585db1beb41f6875fee777134d4/add-npm-global.sh#L37

Any interest in that instead of having to set a flag. Admittedly I didn't look at how extensively the dhcp config is changed in this manner which I probably should do before suggesting changing a behavior in one place without doing so in another.

I may setup an rpm  based system and play around with other ideas this weekend if you don't like that idea or it is too much work to implement properly.

**Comment by Sebastian-Roth** [comment]
Most people install FOG on a system just designated for this one task and therefore never run into the issue. But I can see why you'd like it to be less "aggressive" on overwriting configuration files. On the other hand we definitely have people being used to just run `./installfog.sh -y` to fix things for them if they have tinkered with the configs. I'll reopen this as I can see both point and surely we could tend more to the save corner of not simply overwriting configs. Either unless a command line option is given or if we see that the config has been modified. I'd vote for your inital option number 3. Diffing the files should be possible all distros and files:

We generate all configs in a temporary location/filename, compare it to the existing one and if they match we sure know that no one has touched it or the environment/settings has/have changed or someone had a config already. The only issue we have to come by is that we have a lot of users who don't understand much of linux and would be overwhelmed when asked to decide about a diffed file. So we should have an easy answer mechanism to just accept what "we think is correct".

What do you think Tom?

**Comment by mastacontrola** [comment]
Sorry to rely on you so heavily @Sebastian-Roth and all, but I think it's a great idea to ask the user what they would like to do. Did you get a chance to look into this and all?  We do have the "don't overwrite the exports" option and all but I also see the value in asking the user if they did have custom configurations.

**Comment by mastacontrola** [comment]
No input over the last 10 or so months. I am going to close this. It would be a nice to have feature, but I don't think it should be something that holds up our queue for so long. Maybe later on we can get to this?

**Comment by Sebastian-Roth** [comment]
I came up with a different way of backup up configs and telling the user: https://github.com/FOGProject/fogproject/commit/b4f3295c348164c21e7b37a5eea0b213b840e3e1
